src/main/fc/msp_server_fc.c

```diff
@@ -520,15 +541,21 @@ int mspServerCommandHandler(mspPacket_t *cmd, mspPacket_t *reply)
             sbufWriteU8(dst, FC_VERSION_PATCH_LEVEL);
             break;
 
-        case MSP_BOARD_INFO:
+        case MSP_BOARD_INFO: {
             sbufWriteData(dst, boardIdentifier, BOARD_IDENTIFIER_LENGTH);
 #ifdef USE_HARDWARE_REVISION_DETECTION
             sbufWriteU16(dst, hardwareRevision);
 #else
             sbufWriteU16(dst, 0); // No hardware revision available.
 #endif
-            sbufWriteU8(dst, 0);  // 0 == FC, 1 == OSD, 2 == FC with OSD
+#ifdef OSD
```

Comment:
How does feature mask work for combined board? 

Something like capabilities bits may make sense, but I'am not sure how much do the OSD/FC namespaces overlap.
One  possibility is add OSD/FC addressing to MSP protocol and pretend that there are two/multiple  devices on single bus.